training-service/src/app/pages/mail-form/mail-form.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-mail-form',
+  templateUrl: './mail-form.component.html',
+  styleUrls: ['./mail-form.component.scss']
+})
+export class MailFormComponent implements OnInit {
+
+
+  mailForm = new FormGroup({
+    emailAddress: new FormControl("", [
+      Validators.required,
+      Validators.pattern("^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$")
```

Comment:
Please export ^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$ to a const

Reply:
done.